Content.Server/Animals/Systems/EggLayerSystem.cs

```diff
@@ -39,19 +40,24 @@ public override void Update(float frameTime)
         base.Update(frameTime);
 
         var query = EntityQueryEnumerator<EggLayerComponent>();
+        var now = _timing.CurTime;
         while (query.MoveNext(out var uid, out var eggLayer))
         {
             // Players should be using the action.
             if (HasComp<ActorComponent>(uid))
                 continue;
 
-            eggLayer.AccumulatedFrametime += frameTime;
+            if (now < eggLayer.NextGrowth)
+                continue;
+
+            // Randomize next growth time for more organic egglaying.
+            eggLayer.NextGrowth = now + TimeSpan.FromSeconds(_random.NextFloat(eggLayer.EggLayCooldownMin, eggLayer.EggLayCooldownMax));
 
-            if (eggLayer.AccumulatedFrametime < eggLayer.CurrentEggLayCooldown)
+            if (_mobState.IsDead(uid))
```

Comment:
This is going to eat the remainder after every tick, you should just be adding the timespan to the time on nextgrowth and not currenttime.

Reply:
so I make it `eggLayer.NextGrowth += TimeSpan.FromSeconds(_random.NextFloat(eggLayer.EggLayCooldownMin, eggLayer.EggLayCooldownMax));` right?

Should I do the same in UdderSystem and WoolySystem?